test/suites/integration/expiration_test.go

```diff
@@ -174,4 +181,113 @@ var _ = Describe("Expiration", func() {
 		env.EventuallyExpectCreatedNodeCount("==", 1)
 		env.EventuallyExpectHealthyPodCount(selector, int(numPods))
 	})
+	It("should delete a replacement machine from expiration if it fails to register", func() {
+		// launch a new machine
+		var numPods int32 = 2
+		dep := test.Deployment(test.DeploymentOptions{
+			Replicas: 2,
+			PodOptions: test.PodOptions{
+				ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"app": "inflate"}},
+				PodAntiRequirements: []v1.PodAffinityTerm{{
+					TopologyKey: v1.LabelHostname,
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{"app": "inflate"},
+					}},
+				},
+			},
+		})
+		env.ExpectCreated(dep, nodeTemplate, provisioner)
+		env.EventuallyExpectNodeCount("==", 2)
+
+		// Set a configuration that will not register a machine
+		parameter, err := env.SSMAPI.GetParameter(&ssm.GetParameterInput{
+			Name: aws.String("/aws/service/ami-amazon-linux-latest/amzn-ami-hvm-x86_64-ebs"),
+		})
+		Expect(err).ToNot(HaveOccurred())
+		By(fmt.Sprintf("AMI: %s", *parameter.Parameter.Value))
+		nodeTemplate.Spec.AMISelector = map[string]string{"aws::ids": *parameter.Parameter.Value}
+		env.ExpectCreatedOrUpdated(nodeTemplate)
+
+		// Should see the machine has expired
+		EventuallyWithOffset(1, func(g Gomega) {
+			for _, machine := range env.EventuallyExpectCreatedMachineCount("==", int(numPods)) {
+				g.Expect(env.Client.Get(env, client.ObjectKeyFromObject(machine), machine)).To(Succeed())
+				g.Expect(machine.StatusConditions().GetCondition(v1alpha5.MachineExpired).IsTrue()).To(BeTrue())
+			}
+		}).Should(Succeed())
+
+		// Expect machine To get cordoned
+		env.EventuallyExpectMachineCount("==", 3)
+		nodes := env.EventuallyExpectNodeCount("==", int(numPods))
+		Eventually(func(g Gomega) {
+			g.Expect(len(lo.Filter(nodes, func(node *v1.Node, _ int) bool {
+				g.Expect(env.Client.Get(env, client.ObjectKeyFromObject(node), node))
+				return node.Spec.Unschedulable == true &&
+					lo.ContainsBy(node.Spec.Taints, func(taint v1.Taint) bool { return taint.Key == v1.TaintNodeUnschedulable })
+			}))).To(BeNumerically("==", 1))
+		}).Should(Succeed())
+
+		// Expire should fail and original node should be uncordoned
+		// TODO: reduce timeouts when deprovisioning waits are factored out
+		machineList := &v1alpha5.MachineList{}
+		Eventually(func(g Gomega) {
+			g.Expect(lo.EveryBy(nodes, func(node *v1.Node) bool {
+				g.Expect(env.Client.Get(env, client.ObjectKeyFromObject(node), node))
+				return node.Spec.Unschedulable == false
+			})).To(BeTrue())
+			g.Expect(env.Client.List(env, machineList, client.HasLabels{test.DiscoveryLabel})).To(Succeed())
+			g.Expect(len(machineList.Items)).To(BeNumerically("==", int(numPods)))
+		}).WithTimeout(20 * time.Minute).Should(Succeed())
+		env.EventuallyExpectNodeCount("==", int(numPods))
```

Comment:
I think we _might_ need to do a `Consistently` check here to validate that we aren't proceeding with scale down here. e.g. After the timeout expires and the node is uncordoned, I should ensure that deprovisioning doesn't keep scaling me down